tests/commands/test_store_commands.py

```diff
@@ -1051,6 +1053,290 @@ def test_createlib_path_can_not_write(tmp_path, monkeypatch, store_mock, add_cle
+        ]),
+        call.create_library_revision('testcharm-1', 'lib_id_1', 0, 1, c1, h1),
+        call.create_library_revision('testcharm-1', 'lib_id_2', 0, 1, c2, h2),
+        call.create_library_revision('testcharm-1', 'lib_id_3', 1, 3, c3, h3),
+    ]
+    names = [
+        'charms.testcharm-1.v0.testlib-a',
+        'charms.testcharm-1.v0.testlib-b',
+        'charms.testcharm-1.v1.testlib-b',
+    ]
+    expected = [
+        "Libraries found under lib/charms/testcharm-1: " + str(names),
+        "Library charms.testcharm-1.v0.testlib-a sent to the store with version 0.1",
+        "Library charms.testcharm-1.v0.testlib-b sent to the store with version 0.1",
+        "Library charms.testcharm-1.v1.testlib-b sent to the store with version 1.3",
+    ]
+    records = [rec.message for rec in caplog.records]
+    assert all(e in records for e in expected)
+
+
+def test_publishlib_not_found(caplog, store_mock, tmp_path, monkeypatch):
+    """The indicated library is not found."""
+    caplog.set_level(logging.INFO, logger="charmcraft.commands")
+    monkeypatch.chdir(tmp_path)
+
+    args = Namespace(library='charms.testcharm.v0.testlib')
+    with patch('charmcraft.commands.store.get_name_from_metadata') as mock:
+        mock.return_value = 'testcharm'
+        with pytest.raises(CommandError) as cm:
+            PublishLibCommand('group').run(args)
+
+        assert str(cm.value) == (
+            "The specified library was not found at path lib/charms/testcharm/v0/testlib.py.")
+
+
+def test_publishlib_not_from_current_charm(caplog, store_mock, tmp_path, monkeypatch):
+    """The indicated library to publish does not belong to this charm."""
+    caplog.set_level(logging.INFO, logger="charmcraft.commands")
+    monkeypatch.chdir(tmp_path)
+    factory.create_lib_filepath('testcharm', 'testlib', api=0)
+
+    args = Namespace(library='charms.testcharm.v0.testlib')
+    with patch('charmcraft.commands.store.get_name_from_metadata') as mock:
+        mock.return_value = 'charm2'
+        with pytest.raises(CommandError) as cm:
+            PublishLibCommand('group').run(args)
+
+        assert str(cm.value) == (
+            "The library charms.testcharm.v0.testlib does not belong to this charm 'charm2'.")
+
+
+def test_publishlib_name_from_metadata_problem(store_mock):
+    """The metadata wasn't there to get the name."""
+    args = Namespace(library='charms.testcharm.v0.testlib')
+    with patch('charmcraft.commands.store.get_name_from_metadata') as mock:
+        mock.return_value = None
+        with pytest.raises(CommandError) as cm:
+            PublishLibCommand('group').run(args)
+
+        assert str(cm.value) == (
+            "Can't access name in 'metadata.yaml' file. The 'publish-lib' command needs to "
+            "be executed in a valid project's directory.")
+
+
+def test_publishlib_store_is_advanced(caplog, store_mock, tmp_path, monkeypatch):
+    """The store has a more advanced revision that we expect."""
```

Comment:
Nitpick: perhaps we could use "a revision number higher than what we expect"? "More advanced" seems to imply better technology in the newer revision.

Reply:
Great, changed it.